### ISSUE TITLE
`keybase login` requires upload of private key to keybase?

### ISSUE THREAD

**Issue by timmc** [keybase/keybase-issues#2001]
Am I reading this correctly that I am required to upload my *private key* to keybase in order to log in locally? This is pretty weird. GPG is installed and working, and the gpg agent was even unlocked for one attempt at this (no effect).

```
$ keybase login
How would you like to sign this install of Keybase?

(1) Use an existing device
(2) Use a paper key
(3) Use my Keybase passphrase
(4) Use GPG
Choose a signing option: 3
Your keybase username or email address: timmc
▶ ERROR in Login

Sorry, you have a PGP private key that Keybase doesn't have a copy of.
You need to prove you're you. Any of these will unlock this computer:

   - reset your account and start fresh: https://keybase.io/#account-reset
   - install GPG and put your PGP private key on this machine and try again
   - go back and provision with another device or paper key
```

**Comment by strib** [comment]
@timmc: no, uploading your private key to the Keybase servers is definitely not required.  It's just saying that you need to have the private key locally (the one corresponding to your Keybase PGP public key), on your device, managed by GPG.

Did you try going with option (4)?  GPG agent will definitely need to be unlocked.

**Comment by timmc** [comment]
Ah, so "you have a PGP private key that Keybase doesn't have a copy of" is ambiguous! I read it as "keybase.io doesn't have this key", but you're saying it means "the local keybase client hasn't gotten a copy of the key"?

As for option 4, I already tried that and was told in https://github.com/keybase/keybase-issues/issues/1991 that it's not supported yet.

**Comment by strib** [comment]
Well, that comment reads the way you think it reads, but it doesn't follow from there that it's required for you to upload it.  (4) should work.  On #1991, @maxtaco was saying (I think) that you have to type in your Keybase account passphrase to establish the session with the Keybase server.  Once that session is established, the Keybase binary will use your local PGP key to sign your new device into your sigchain.

Have you tried entering your password as part of (4), and not hitting cancel?

**Comment by timmc** [comment]
OK, I think I see now. Now that I understand the purpose of it, I gave it a shot. (4) GPG followed by (1) use GPG commands allowed me to sign in my laptop and generate a paper key. Thanks!

I still think the language "a PGP private key that Keybase doesn't have a copy of" is ambiguous in a bad way.

**Comment by strib** [comment]
Great, glad it worked!  @patrickxb @maxtaco, what do you guys think about improving the wording?
